packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx

```diff
@@ -291,20 +368,20 @@ function ChartsXAxis(inProps: ChartsXAxisProps) {
               />
             )}
 
-            {formattedValue !== undefined && showTickLabel && (
+            {tickLabel !== undefined && showTickLabel && (
               <TickLabel
                 x={xTickLabel}
                 y={yTickLabel}
                 data-testid="ChartsXAxisTickLabel"
                 {...axisTickLabelProps}
-                text={formattedValue.toString()}
+                text={tickLabel}
               />
             )}
           </g>
         );
       })}
 
-      {label && (
+      {label && isClient && (
```

Comment:
I assume you introduced it before getting the fallback for serve side

Reply:
Yeah. The label's position depends on its height, but on the server-side the height is always 0, so it would be incorrectly positioned:

https://github.com/user-attachments/assets/3073cd3e-b8bc-4e85-b89c-815ed20b34ad